src/agent/src/device.rs

```diff
@@ -799,7 +797,12 @@ pub fn update_device_cgroup(spec: &mut Spec) -> Result<()> {
         linux.resources = Some(LinuxResources::default());
     }
 
-    let resources = linux.resources.as_mut().unwrap();
+    let resources = linux
+        .resources
+        .as_mut()
+        .ok_or("failed to get Linux resources")
+        .map_err(|e| anyhow!(e))?;
+
```

Comment:
`unwrap()` is appropriate here, since we literally just handled the case where `linux.resources` was `None` by populating it with a default.  It wouldn't surprise me if the compiler already optimzed the `unwrap()` to a no-op..

Reply:
Looks good.